plugin/src/main/java/org/opensearch/ml/action/agents/DeleteAgentTransportAction.java

```diff
@@ -116,4 +177,22 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Delete
     boolean isSuperAdminUserWrapper(ClusterService clusterService, Client client) {
         return RestActionUtils.isSuperAdminUser(clusterService, client);
     }
+
+    private void handleDeleteResponse(
+        DeleteDataObjectResponse response,
+        Throwable throwable,
+        String agentId,
+        ActionListener<DeleteResponse> actionListener
+    ) {
+        if (throwable != null) {
+            Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
+            log.error("Failed to delete ML Agent : {}", agentId, cause);
+            actionListener.onFailure(cause);
+        } else {
+            log.info("Agent deletion result: {}, agent id: {}", response.deleted(), response.id());
+            DeleteResponse deleteResponse = new DeleteResponse(response.shardId(), response.id(), 0, 0, 0, response.deleted());
```

Comment:
This seems a breaking change by harding code `0, 0, 0`. Is it possible to record real values in `DeleteDataObjectResponse`, so we can get the real values just like `response.shardInfo()`

Reply:
@ylwu-amzn yes, I've noted this and had discussion with @dhrubo-os and @arjunkumargiri ... will fix Delete, Update, and Get to process full response similar to #2587. 

More discussion here: https://github.com/opensearch-project/ml-commons/pull/2587/files#r1651490825